source/distributed/tria.cc

```diff
@@ -3506,7 +3506,10 @@ namespace parallel
             cell->clear_refine_flag ();
             cell->clear_coarsen_flag ();
           }
-
+      this->prepare_coarsening_and_refinement ();
+      // If flags come out after mesh smoothing on ghost and artificial cells,
+      // we should keep them because they are resulted from smoothing of locally
+      // owned active cells. So no cleaning is needed here.
```

Comment:
On 10/12/2015 03:55 PM, Lei Qiao wrote:
> Oh, I got your point. But I still believe it's better to keep the flags after
> smoothing. Because the flags are meaningful rather than random or undefined.
> Anyway there is also nobody relay on this piece of information. So for now
> this is not a critical issue.

Not immediately, but you should probably make sure that once you get out of 
this function, no flags remain set.